src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -161,6 +161,14 @@ void BulletPhysicsManager::stepPhysics(double dt) {
   int numSubStepsTaken =
       bWorld_->stepSimulation(dt, /*maxSubSteps*/ 10000, fixedTimeStep_);
   worldTime_ += numSubStepsTaken * fixedTimeStep_;
+
+  // Manually sync the motionstates for all DYNAMIC objects. KINEMATIC and
+  // STATIC sync should always be one-way.
+  for (auto& objectItr : existingObjects_) {
```

Comment:
This makes sense. Can we avoid giving Bullet our btMotionState interface in BulletRigidObject::constructAndAddRigidBody, so that Bullet doesn't bother setting wrong transforms at every step?